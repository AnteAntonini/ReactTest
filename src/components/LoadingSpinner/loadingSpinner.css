.loading-spinner-wrapper {
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100vh;
}

.loading-spinner {
  width: 50px;
  height: 50px;
  border: 10px solid var(--color-martian-lightgray);
  border-top: 10px solid var(--color-martian-dark);
  border-radius: 50%;
  animation: spinner 1.5s linear infinite;
}

@keyframes spinner {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
}
