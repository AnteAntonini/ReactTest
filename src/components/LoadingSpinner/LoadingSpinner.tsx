import "./loadingSpinner.css";
import { useEffect } from "react";

export default function LoadingSpinner({ message }: { message: string }) {
  useEffect(() => {
    console.log(`${message} Loading Spinner`);
  }, []);

  return (
    <div className="loading-spinner-wrapper">
      <div className="loading-spinner"></div>
    </div>
  );
}
