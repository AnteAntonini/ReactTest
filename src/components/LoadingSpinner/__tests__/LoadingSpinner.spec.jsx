import { render } from "@testing-library/react";
import LoadingSpinner from "../LoadingSpinner";

describe("LoadingSpinner Component", () => {
  it("renders the loading spinner", () => {
    const { container } = render(<LoadingSpinner />);

    // Check if the spinner container and loading spinner elements are present
    const spinnerContainer = container.querySelector(
      ".loading-spinner-wrapper"
    );
    const loadingSpinner = container.querySelector(".loading-spinner");

    expect(spinnerContainer).toBeInTheDocument();
    expect(loadingSpinner).toBeInTheDocument();
  });
});

describe("Test snapshot", () => {
  it("Component should match snapshot", () => {
    const { asFragment } = render(<LoadingSpinner />);

    expect(asFragment()).toMatchSnapshot();
  });
});
