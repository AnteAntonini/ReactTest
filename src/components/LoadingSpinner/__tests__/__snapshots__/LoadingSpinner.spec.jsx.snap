// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test snapshot Component should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="spinner-container"
  >
    <div
      class="loading-spinner"
    />
  </div>
</DocumentFragment>
`;
